app/ModelFunctions/PhotoFunctions.php

```diff
@@ -286,10 +315,19 @@ public function add(array $file, $albumID_in = 0)
 			}
 		}
 
-		$info = $this->metadataExtractor->extract($path, $mimeType);
+		if ($kind == 'RAW') {
+			$info = $this->metadataExtractor->bare();
+			$this->metadataExtractor->size($info, $path);
+			$this->metadataExtractor->validate($info);
+			$info['type'] = 'raw';
+		} else {
+			$info = $this->metadataExtractor->extract($path, $mimeType);
+		}
 
 		// Use title of file if IPTC title missing
-		if ($info['title'] === '') {
+		if ($kind == 'RAW') {
+			$info['title'] = substr(basename($file['name']), 0, 30);
```

Comment:
I expected as much based on the earlier code but I still don't know why for RAWs you choose to keep the extension in the title?

Reply:
Because I would rather like to know the kind of file I'm dealing with directly by looking at its title rather than looking at the mimetype.

Reply:
I'm not sure about this either... I can see advantages, but...